src/saml2/sigver.py

```diff
@@ -957,7 +957,10 @@ def sign_statement(self, statement, node_name, key_file, node_id, id_attr):
 
         xml = xmlsec.parse_xml(statement)
         signed = xmlsec.sign(xml, key_file)
-        return lxml.etree.tostring(signed, xml_declaration=True)
+        if six.PY2:
+            return lxml.etree.tostring(signed, xml_declaration=True, encoding='UTF-8')
```

Comment:
From the `lxml` documentation for `tostring` function:
```
Defaults to ASCII encoding without XML declaration.  This
behaviour can be configured with the keyword arguments 'encoding'
(string) and 'xml_declaration' (bool).  Note that changing the
encoding to a non UTF-8 compatible encoding will enable a
declaration by default.

You can also serialise to a Unicode string without declaration by
passing the name ``'unicode'`` as encoding (or the ``str`` function
in Py3 or ``unicode`` in Py2).  This changes the return value from
a byte string to an unencoded unicode string.
```

---

I think this line should be good for both python2 and python3.
In the not-`PY2` case, you're omitting the xml declaration. Is that on purpose?